### PR TITLE
authorize subresources

### PR DESCRIPTION
**Pull request by deads2k**

### CODE REVIEW

**Comment by deads2k** [comment]
[test]

**Comment by deads2k** [comment]
@liggitt I'm feeling optimistic about the test.  This turned out to be surprisingly straightforward compared to moving the implementation to the api installer.

File: Godeps/_workspace/src/github.com/GoogleCloudPlatform/kubernetes/pkg/apiserver/handlers.go [hunk 1]
```diff
@@ -211,6 +211,8 @@ type APIRequestInfo struct {
 	Namespace  string
 	// Resource is the name of the resource being requested.  This is not the kind.  For example: pods
 	Resource string
+	// Subresource is the name of the subresource being requested.  This is not the kind or the resource.  For example: status for a pods/pod-name/status
```
Comment (liggitt) [inline Godeps/_workspace/src/github.com/GoogleCloudPlatform/kubernetes/pkg/apiserver/handlers.go]: typo, "kind of the resource"?

Comment (deads2k) [inline Godeps/_workspace/src/github.com/GoogleCloudPlatform/kubernetes/pkg/apiserver/handlers.go]: Not a typo.  It is not the kind or the resource.

File: Godeps/_workspace/src/github.com/GoogleCloudPlatform/kubernetes/pkg/apiserver/handlers.go [hunk 2]
```diff
@@ -312,12 +314,10 @@ func (r *APIRequestInfoResolver) GetAPIRequestInfo(req *http.Request) (APIReques
 	// URL forms: /namespaces/{namespace}/{kind}/*, where parts are adjusted to be relative to kind
 	if currentParts[0] == "namespaces" {
 		if len(currentParts) < 3 {
```
Comment (liggitt) [inline Godeps/_workspace/src/github.com/GoogleCloudPlatform/kubernetes/pkg/apiserver/handlers.go]: Could this be simplified to:
```
if len(currentParts) > 1 {
  requestInfo.Namespace = currentParts[1]
  if len(currentParts) > 2 {
    currentParts = currentParts[2:]
  }
}
```

**Comment by liggitt** [comment]
nits, but LGTM

File: Godeps/_workspace/src/github.com/GoogleCloudPlatform/kubernetes/pkg/apiserver/handlers.go (hunk 2 above)
Comment (deads2k) [inline Godeps/_workspace/src/github.com/GoogleCloudPlatform/kubernetes/pkg/apiserver/handlers.go]: ok

**Comment by deads2k** [comment]
comments addressed.  [merge]

**Comment by deads2k** [comment]
@ncdc exec transient: https://ci.openshift.redhat.com/jenkins/job/merge_pull_requests_openshift3/1473/

re[merge]

**Comment by ncdc** [comment]
Thanks, I'm aware of the exec issue and have a hunch as to the root cause. I'll see about a fix tomorrow.

**Pull request merged**